bot/exts/evergreen/splitify.py

```diff
@@ -0,0 +1,130 @@
+import math
+import random
+from io import BytesIO
+
+import discord
+from PIL import Image
+from discord.ext import commands
+
+
+MAX_SQUARES = 193_600
+
+
+class Splitify(commands.Cog):
+    """Splits your avatar in x squares!"""
+
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+
+    @staticmethod
+    def split_image(img: Image, squares: int) -> list:
+        """Splits the image into x squares."""
```

Comment:
Could you explain the algorithm you used for solving this task in the doc-string?

Reply:
Is there an issue with it or are you just personally interested?

Reply:
It would help one going through the code and understanding how it is done.

Say, one contributor wants to refactor this code and change the implementation a bit, he would need to understand this code for doing that, when you put the doc-strings in there, it would help them do it.

Reply:
Well may I explain myself then.

1. I get the width and the height of the Image passed to the function.

2. I get the root of a number of squares passed which I call xy. Reason: if let's say 25 squares were passed, that is the total squares (split pieces) that the image is supposed to be split into. As you know, a 2D shape has a height and a width, and for this case I think of it as rows and columns. Rows multiplied by columns is equal to the passed squares. To get rows and columns, since in this case each square is identical, rows are equal to columns and I treat the Image as a square-shaped, I get the root out of the squares passed.

3. Now width and height are both of the original Image, Discord PFP, so when it comes to forming the squares (split pieces), I divide the original image's height and width by the xy. In a case of 25 squares passed, xy would be 5, so if an image was 250x300, x_frac would be 50 and y_frac - 60. Note: x_frac stands for a fracture of width. Reason I called it so is because it is shorter to use x for width in mind and then I just used half of the word fracture, same applies to y_frac, just height instead of width. x_frac and y_frac are width and height of a single square (split piece).

4. With `left, top, right, bottom, = 0, 0, x_frac, y_frac`, I set these variables to create the initial square (split piece). Explanation: all of these 4 variables start at the top left corner of the Image, by adding value to right and bottom, I am creating the initial square (split piece).

5. In the for loop, I keep adding those squares (split pieces) in row and once `(index + 1) % xy == 0` is True, I add to top and bottom to lower them and reset right and left to recreate the initial space between them, forming a square (split piece), I also add the newly created square (split piece) into the new_imgs list where I store them. I keep repeating this process till all 25 squares get added to the list.

Hopefully this is comprehendible.

Reply:
I did understand the code originally but Thanks for the awesome explanation, what I was asking for to include this explanation in your code as comments and doc-strings so the algorithm can be much clear to other contributors.

Reply:
Alright, I'll do that.